src/components/textAreas/textAreaRichText/textAreaRichTextActions.test.tsx

```diff
@@ -0,0 +1,124 @@
+import { fireEvent, render, screen } from '@testing-library/react';
+import type { ChainedCommands, Editor } from '@tiptap/react';
+import { IconType } from '../../icon';
+import { TextAreaRichTextActions, type ITextAreaRichTextActionsProps } from './textAreaRichTextActions';
+
+describe('<TextAreaRichTextActions /> component', () => {
+    const windowPromptMock = jest.spyOn(window, 'prompt');
+
+    afterEach(() => {
+        windowPromptMock.mockReset();
+    });
+
+    const createTestComponent = (props?: Partial<ITextAreaRichTextActionsProps>) => {
+        const completeProps: ITextAreaRichTextActionsProps = {
+            editor: null,
+            ...props,
+        };
+
+        return <TextAreaRichTextActions {...completeProps} />;
+    };
+
+    it('renders the text-area actions', () => {
+        render(createTestComponent());
+        expect(screen.getAllByRole('button')).toHaveLength(6);
+    });
+
+    it('renders the actions as disabled when the isDisabled property is set to true', () => {
+        const isDisabled = true;
+        render(createTestComponent({ isDisabled }));
+        screen.getAllByRole<HTMLButtonElement>('button').forEach((button) => expect(button.disabled).toBeTruthy());
+    });
+
+    it('renders the set link action instead of the unset one when current active node is not a link', () => {
+        const editor = { isActive: () => false } as unknown as Editor;
+        render(createTestComponent({ editor }));
+        expect(screen.getByTestId(IconType.WYSIWYG_LINK_SET)).toBeInTheDocument();
+        expect(screen.queryByTestId(IconType.WYSIWYG_LINK_UNSET)).not.toBeInTheDocument();
+    });
+
+    it('renders the unset link action instead of the set one when current active node is not a link', () => {
+        const editor = { isActive: () => true } as unknown as Editor;
```

Comment:
This is the fun part of testing, isn't it? This tests the unit properly but then the integration of the component into the whole as well needs testing. Sigh